File: src/test/java/com/raychenon/leetcode/PartitionLabelsTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import java.util.List;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 4/9/2020
 * https://leetcode.com/problems/partition-labels
 */
public class PartitionLabelsTest {

    private static PartitionLabels instance;

    @BeforeClass
    public static void setUp() {
        instance = new PartitionLabels();
    }

    @Test
    public void ababcbacadefegdehijhklijTest() {
        assertEquals(List.of(9, 7, 8), instance.partitionLabels("ababcbacadefegdehijhklij"));
    }

    @Test
    public void ababcbadefegdehijhklijopopTest() {
        assertEquals(List.of(7, 7, 8, 4), instance.partitionLabels("ababcbadefegdehijhklijopop"));
    }

    @Test
    public void abcdefghijklTest() {
        assertEquals(List.of(1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1), instance.partitionLabels("abcdefghijkl"));
    }


}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/PartitionLabelsTest.kt
package com.raychenon.kotlin.leetcode

import junit.framework.TestCase
import org.junit.Test

/**
 * User: raychenon
 * Date: 5/9/2020
 * https://leetcode.com/problems/partition-labels
 */
class PartitionLabelsTest {

    @Test
    fun ababcbacadefegdehijhklijTest() {
        evaluatePartitionLabels(listOf<Int>(9, 7, 8), ("ababcbacadefegdehijhklij"))
    }

    @Test
    fun ababcbadefegdehijhklijopopTest() {
        evaluatePartitionLabels(listOf<Int>(7, 7, 8, 4), ("ababcbadefegdehijhklijopop"))
    }

    @Test
    fun abcdefghijklTest() {
        evaluatePartitionLabels(listOf<Int>(1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1), ("abcdefghijkl"))
    }

    private fun evaluatePartitionLabels(expected: List<Int>, S: String): Unit {
        TestCase.assertEquals(expected, PartitionLabels.partitionLabelsOriginalSolution(S))
    }
}